martin/src/config/file/main.rs

```diff
@@ -218,15 +222,32 @@ impl Config {
         #[cfg(feature = "pmtiles")]
         let pmtiles_cache = cache_config.create_pmtiles_cache();
 
+        let (tiles, warnings) = self
+            .resolve_tile_sources(
+                &resolver,
+                #[cfg(feature = "pmtiles")]
+                pmtiles_cache,
+            )
+            .await?;
+
+        if !warnings.is_empty() {
+            if self.fail_on_source_resolution_warnings {
+                for warning in &warnings {
+                    log::error!("Error while resolving source: {}", warning);
+                }
+                return Err(MartinError::TileResolutionWarningsIssued);
+            }
+            log::warn!(
+                "Warnings issued during source resolution. These sources will not be available:"
+            );
+            for warning in &warnings {
+                log::warn!("- {}", warning);
+            }
+        }
```

Comment:
nit: I think writing it this way might be more elegant.
Not sure, current version is likely fine too.

```suggestion
        #[cfg(feature = "_tiles")]
        let tiles = {
            let (tiles, warnings) = self
                .resolve_tile_sources(
                    &resolver,
                    #[cfg(feature = "pmtiles")]
                    pmtiles_cache,
                )
                .await?;
            if !warnings.is_empty() {
                OnInvalid::handle_tiles(warnings)?;
            }
            tiles
        }
```

Reply:
I like it. Keeps more of the model & policy in the same place.